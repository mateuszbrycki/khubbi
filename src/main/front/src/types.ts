import {DateTimeFormatter, LocalDateTime, ZonedDateTime, ZoneId} from "js-joda";

export class EventDate {
    value: ZonedDateTime

    public constructor(value: ZonedDateTime) {
        this.value = value
    }

    public toString(): string {
        return this.value.toString();
    }

    public format(format: string): string {
        return this.value.format(DateTimeFormatter.ofPattern(format))
    }

    public toISODateTime(): string {
        return this.value.format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
    }

    public atUTC(): EventDate {
        return new EventDate(this.value.withZoneSameLocal(ZoneId.UTC))
    }

    public static now(): EventDate {
        return new EventDate(ZonedDateTime.now());
    }

    public static ofDateAndTime(date: string): EventDate {
        return new EventDate(LocalDateTime.parse(date)
            .atZone(ZoneId.SYSTEM));
    }

    public static ofZoneDateAndTime(date: string): EventDate {
        return new EventDate(ZonedDateTime.parse(date));
    }

}

export class Photo {
    readonly date: EventDate;
    readonly id: string;
    readonly description: string;
    readonly photo: string;

    constructor(id: string, date: EventDate, description: string, photo: string) {
        this.date = date;
        this.id = id;
        this.description = description;
        this.photo = photo;
    }
}

export class Note {
    readonly date: EventDate;
    readonly id: string;
    readonly note: string;

    constructor(id: string, date: EventDate, note: string) {
        this.date = date;
        this.id = id;
        this.note = note;
    }
}

export type Event = Photo | Note

export interface User {
    readonly id: string;
    readonly email: string;
    readonly createdAt: string;
}

export class EventForms {
    public static readonly NOTE = new EventForms('Note');
    public static readonly PICTURE = new EventForms('Picture');
    public static readonly EVENT = new EventForms('Event');
    public static readonly EXPENSE = new EventForms('Expense');
    public static readonly REMINDER = new EventForms('Reminder');

    private constructor(public readonly name: string) {
    }
}