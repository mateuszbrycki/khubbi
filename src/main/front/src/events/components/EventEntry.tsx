import React from "react";
import {Event} from "../../types";
import Card from 'react-bootstrap/Card';

export interface EventEntryProps {
    readonly event: Event

}

export interface EventEntryActionProps {
}

const EventEntry: React.FC<EventEntryProps & EventEntryActionProps> = (props) => {
    const {event} = props

    return <Card className="p-0 mb-3">
        <Card.Body>
            <Card.Text>
                {event.note}
            </Card.Text>
        </Card.Body>
        <Card.Footer>
            {event.date.format("dd-MM-yyyy, HH:mm")}
        </Card.Footer>
    </Card>
}

export default EventEntry