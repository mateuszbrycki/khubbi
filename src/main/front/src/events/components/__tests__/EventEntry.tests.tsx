import ShallowRenderer from "react-test-renderer/shallow";
import {EventDate, Note, Photo} from "../../../types";
import EventEntry from "../EventEntry";

it('renders event entry with note ', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<EventEntry event={new Note(
        "event-1-id",
        EventDate.ofDateAndTime("2024-07-12T20:00").atUTC(),
        "event-1-note")
    }
    />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
});

it('renders event entry with photo ', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<EventEntry event={new Photo(
        "event-1-id",
        EventDate.ofDateAndTime("2024-07-12T20:00").atUTC(),
        "event-1-description",
        "event-1-photo-url",
    )
    }
    />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
});

it('renders event entry with no content when unrecognized entry ', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<EventEntry event={{
        note: "event-1",
        id: "event-1",
        date: EventDate.ofDateAndTime("2024-07-12T20:00").atUTC()
    }}
    />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
});

