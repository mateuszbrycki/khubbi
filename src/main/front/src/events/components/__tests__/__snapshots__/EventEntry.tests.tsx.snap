// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders event entry with no content when unrecognized entry  1`] = `
<React.Fragment>
  <React.Fragment />
</React.Fragment>
`;

exports[`renders event entry with note  1`] = `
<React.Fragment>
  <Card
    className="p-0 mb-3"
  >
    <CardHeader>
      <ForwardRef
        className="me-2"
        color="currentColor"
        size="1em"
        title={null}
      />
    </CardHeader>
    <CardBody>
      <CardText>
        event-1-note
      </CardText>
    </CardBody>
    <CardFooter>
      12-07-2024, 20:00
    </CardFooter>
  </Card>
</React.Fragment>
`;

exports[`renders event entry with photo  1`] = `
<React.Fragment>
  <Card
    className="p-0 mb-3"
  >
    <CardHeader>
      <ForwardRef
        className="me-2"
        color="currentColor"
        size="1em"
        title={null}
      />
       
      event-1-photo-url
    </CardHeader>
    <CardBody>
      <CardText>
        event-1-description
      </CardText>
    </CardBody>
    <CardFooter>
      12-07-2024, 20:00
    </CardFooter>
  </Card>
</React.Fragment>
`;
